chain/network/src/rate_limits/messages_limits.rs

```diff
@@ -0,0 +1,444 @@
+//! This module facilitates the initialization and the storage
+//! of rate limits per message.
+
+use std::collections::HashMap;
+
+use enum_map::{enum_map, EnumMap};
+use near_async::time::Instant;
+
+use crate::network_protocol::{PeerMessage, RoutedMessageBody};
+
+use super::token_bucket::{TokenBucket, TokenBucketError};
+
+/// Object responsible to manage the rate limits of all network messages
+/// for a single connection/peer.
+#[derive(Default)]
+pub struct RateLimits {
+    buckets: EnumMap<RateLimitedPeerMessageKey, Option<TokenBucket>>,
+}
+
+impl RateLimits {
+    /// Creates all buckets as configured in `config`.
+    /// See also [TokenBucket::new].
+    pub fn from_config(config: &Config, start_time: Instant) -> Self {
```

Comment:
Maybe it would make sense to print some kind of warning if the override is lower than the default.

Reply:
Changing this configuration is kinda tricky, I don't see many operators doing that - they need to know what they are doing. Is emitting warning for config changes a pattern we use in other places?

Reply:
Oh hmm why do you say the configuration is tricky to change? If I want to bump up my receive rate for a certain message type, do I need to do more than adding a line to the config.json in the right place?

To answer your question, I didn't find an example of warns specifically but there are some examples [here](https://github.com/near/nearcore/blob/fc3c82c771e2612f1cd2c8f896303dfcf5aa9d9b/chain/network/src/config.rs#L248-L278) and [here](https://github.com/near/nearcore/blob/fc3c82c771e2612f1cd2c8f896303dfcf5aa9d9b/chain/network/src/config.rs#L458-L505) of checks on config values. Of course a `bail!` is impossible to miss (node won't start) while a warning may not even be seen, so if it's not super easy to add I wouldn't waste time on it.

Reply:
Yes, it's easy once you know what to do, a simple addition to config.json. 
It's just that if I put myself in the shoes of an average operator it might non-trivial to know what values to use. A knowledge problem rather than a technical one.

Personally, I think we can go without the warning and keep the `bail!` if validation fails, for now at least.